Makefile

```
@@ -99,10 +99,9 @@ watch-tests: ## Watch tests and run on change
 
 .PHONY: freeze-requirements
 freeze-requirements: ## Pin all requirements including sub dependencies into requirements.txt
-	pip install --upgrade pip-tools
-	pip-compile requirements.in
+	uv pip compile requirements.in
```

https://docs.astral.sh/uv/pip/compile/#locking-requirements

> Note by default the uv pip compile output is just displayed and --output-file / -o argument is needed to write to a file.

```suggestion
	uv pip compile requirements.in -o requirements.txt
```

(and the same for the requirements_for_test line below)